### PR TITLE
[MDB-27106] Use `system.remote_data_paths` table for cleaning S3

### PR DESCRIPTION
**Pull request by aalexfvk**
Use `system.remote_data_paths` table for enumerating objects known to ClickHouse. 
Enumerated objects from S3 is placed into table in ClickHouse. And selecting of objects presented only on S3 is performed by ANTIJOIN query with `system.remote_data_paths` table.

TODO (not implemented in the PR): 
1) Counting total size of objects
2) Improve embedded ClickHouse client to make streaming INSERTs.
3) Add additional checks for incomplete deletion (`requests` might not throw an error while disconnecting while stream request)

Note:
PR is big due to updating of `poetry.lock`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -185,7 +185,7 @@ test-unit: install-deps
 test-integration: install-deps build-python-packages	
 	cd $(TESTS_DIR)
 	$(POETRY) run $(PYTHON) -m env_control create
-	$(POETRY) run behave --show-timings --stop -D skip_setup --junit $(BEHAVE_ARGS)
```
Comment (Alex-Burmak) [inline Makefile:188]: Do we still need to execute `env_control create` after removing `-D skip_setup`?

File: ch_tools/chadmin/cli/object_storage_group.py [hunk 2]
```diff
@@ -129,146 +74,180 @@ def object_storage_group(ctx: Context, config_path: Path, disk_name: str) -> Non
...
-
-    with dump_writer(compressed, dump_file) as writer:
-        object_key_to_metadata = collect_metadata(
-            get_disk_metadata_paths(disk_conf.name)
+    listing_table = LISTING_TABLE_PREFIX + disk_conf.name
+    # Create listing table for storing paths from object storage
+    try:
+        execute_query(
+            ctx,
+            f"CREATE TABLE IF NOT EXISTS {listing_table} (obj_path String) ENGINE MergeTree ORDER BY obj_path",
```
Comment (Alex-Burmak) [inline ch_tools/chadmin/cli/object_storage_group.py:127]: In what database this table will be created? Suggest to make it configurable through ch-tools configuration file (https://github.com/yandex/ch-tools/blob/master/ch_tools/common/config.py).

File: ch_tools/chadmin/cli/object_storage_group.py [hunk 3]
```diff
@@ -129,146 +74,180 @@ def object_storage_group(ctx: Context, config_path: Path, disk_name: str) -> Non
...
-        file = (
-            obj.key
-            for _, obj in s3_object_storage_iterator(
-                disk_conf, path_prefix=prefix, skip_ignoring=True
-            )
-        )
+    for _, obj in s3_object_storage_iterator(
+        ctx.obj["disk_configuration"], object_name_prefix=prefix
+    ):
+        if obj.last_modified > now - to_time:
```
Comment (MikhailBurdukov) [inline ch_tools/chadmin/cli/object_storage_group.py:215]: In the knobs description:
```
"Objects with a modification time falling interval [now - from_time, now - to_time] are considered.
```

But here `now-to_time` is the lower bound.

File: tests/features/object_storage.feature [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    CREATE TABLE IF NOT EXISTS test.table_s3_01 ON CLUSTER '{cluster}' (n Int32)
+    ENGINE = ReplicatedMergeTree('/tables/table_s3_01', '{replica}') ORDER BY n
+    SETTINGS storage_policy = 'object_storage';
+
+    INSERT INTO test.table_s3_01 (n) SELECT number FROM system.numbers LIMIT 10;
+    """
+    Then S3 contains greater than 0 objects
+
+  @require_version_22.8
```
[the remark this answers is not in the document]
Comment (aalexfvk) [reply tests/features/object_storage.feature]: Removed require decorators

File: tests/features/object_storage.feature [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    """
+
+  @require_version_22.8
+  Scenario: Clean many orphaned objects
+    When we put 100 objects in S3
+    """
+      bucket: cloud-storage-test
+      path: /data/orpaned_object.tsv
+      data: '1'
+    """
```
[the remark this answers is not in the document]
Comment (aalexfvk) [reply tests/features/object_storage.feature]: Changed to more clear style

File: ch_tools/chadmin/cli/object_storage_group.py [hunk 6]
```diff
@@ -55,60 +35,25 @@ def get_disk_metadata_paths(disk_name: str) -> List[Path]:
...
-@object_storage_group.command("list")
-@option(
-    "-o",
-    "--orphaned",
-    "orphaned",
-    is_flag=True,
-    default=False,
-    help="List objects that are not referenced in the metadata.",
-)
+@object_storage_group.command("clean")
```
[the remark this answers is not in the document]
Comment (aalexfvk) [reply ch_tools/chadmin/cli/object_storage_group.py:50]: This command is useless for now. Before it was intended for generating a list of orpahned objects to stdout for reading by another cleaning script. But now we do this in one `clean` command.
Nevertheless, we can reproduce this functionality for now by `chadmin object-storage clean --keep-paths --dry-run`.

File: ch_tools/chadmin/cli/object_storage_group.py (hunk 3 above)
Comment (aalexfvk) [reply ch_tools/chadmin/cli/object_storage_group.py:238]: It is an upper bound of an interval
https://github.com/yandex/ch-tools/blob/3dfdae2f8f0d5540ae202a964cb8572496dc9cca/ch_tools/chadmin/cli/object_storage_group.py#L238-L239

Actually we filter objects newer than `now - 24h` point in time and pass objects created more than 24h ago.

**Review by MikhailBurdukov — COMMENTED** [review]
the rest looks good

**Review by Alex-Burmak — APPROVED** [review]

File: ch_tools/chadmin/cli/object_storage_group.py [hunk 7]
```diff
@@ -129,146 +78,199 @@ def object_storage_group(ctx: Context, config_path: Path, disk_name: str) -> Non
...
-    Clean up needless S3 objects.
+    Traverse S3 disk's bucket and put object names to the ClickHouse table.
     """
-    disk_conf: S3DiskConfiguration = ctx.obj["disk_configuration"]
-
-    if compressed:
-        file = GzipFile(fileobj=file)
-    file = TextIOWrapper(file)
+    obj_paths_batch = []
+    counter = 0
```
[the remark this answers is not in the document]
Comment (aalexfvk) [reply ch_tools/chadmin/cli/object_storage_group.py:232]: Added info message with it

**Pull request merged**